src/test_fixture/event_gen.rs

```diff
@@ -241,8 +254,12 @@ impl<R: Rng> Iterator for EventGenerator<R> {
     type Item = TestRawDataRecord;
 
     fn next(&mut self) -> Option<Self::Item> {
-        const USERS_IN_FLIGHT: usize = 10;
-        while self.users.len() < USERS_IN_FLIGHT {
+        let number_of_users_in_flight = if self.config.single_user_mode {
+            1
+        } else {
+            usize::try_from(self.config.number_of_users_in_flight.get()).unwrap()
+        };
+        while self.users.len() < number_of_users_in_flight {
```

Comment:
Maybe we move this into the constructor, and make it possible to configure it. Then this "replace" logic can just maintain self.users at a constant length.